File: capture.mjs
import { chromium } from "playwright";
import fs from "fs";

const OUT = "/home/user/afrigen/shots";
fs.mkdirSync(OUT, { recursive: true });
const BASE = "http://localhost:4200";
const PASS = "afrigen2026";

const roles = {
  client:   { email: "client@afrigen.africa",   tabs: [["pipeline","/app"],["new","/app/new"],["cargo","/app/cargo"]] },
  supplier: { email: "supplier@afrigen.africa", tabs: [["fleet","/app"],["vault","/app/vault"],["breakdown","/app/breakdown"],["cargobids","/app/cargo"]] },
  field:    { email: "field@afrigen.africa",    tabs: [["audits","/app"],["border","/app/border"]] },
  admin:    { email: "admin@afrigen.africa",    tabs: [["overview","/app"],["verify","/app/verify"],["ledger","/app/ledger"]] },
};

async function login(page, email) {
  await page.goto(BASE, { waitUntil: "networkidle" });
  await page.evaluate(() => localStorage.clear());
  await page.goto(BASE, { waitUntil: "networkidle" });
  // ensure signin mode
  const signinBtn = page.getByRole("button", { name: "Sign in" }).first();
  await signinBtn.click().catch(() => {});
  await page.locator('input[type="email"]').fill(email);
  await page.locator('input[type="password"]').fill(PASS);
  await page.getByRole("button", { name: /Sign in/i }).last().click();
  await page.waitForTimeout(2500);
}

const wait = (p, ms = 1600) => p.waitForTimeout(ms);

const browser = await chromium.launch();
const ctx = await browser.newContext({ viewport: { width: 1920, height: 1080 }, deviceScaleFactor: 1 });
const page = await ctx.newPage();

// 1. Auth screen (clean, before login)
await page.goto(BASE, { waitUntil: "networkidle" });
await page.evaluate(() => localStorage.clear());
await page.reload({ waitUntil: "networkidle" });
await wait(page);
await page.screenshot({ path: `${OUT}/00-auth.png` });
console.log("shot 00-auth");

for (const [role, cfg] of Object.entries(roles)) {
  await login(page, cfg.email);
  for (const [name, href] of cfg.tabs) {
    await page.goto(BASE + href, { waitUntil: "networkidle" });
    await wait(page);
    const file = `${role}-${name}.png`;
    await page.screenshot({ path: `${OUT}/${file}` });
    console.log("shot", file);
  }
  // client: capture a contract detail (the active cross-border hero contract)
  if (role === "client") {
    await page.goto(BASE + "/app", { waitUntil: "networkidle" });
    await wait(page);
    const link = page.getByText(/Open|View|Details/i).first();
    if (await link.count()) {
      await link.click().catch(() => {});
      await wait(page, 2000);
      await page.screenshot({ path: `${OUT}/client-contract.png` });
      console.log("shot client-contract");
    }
  }
}

await browser.close();
console.log("DONE");

File: capture.py
import os
from playwright.sync_api import sync_playwright

OUT = "/home/user/afrigen/shots"
os.makedirs(OUT, exist_ok=True)
BASE = "http://localhost:4200"
PASS = "afrigen2026"

roles = {
    "client":   {"email": "client@afrigen.africa",   "tabs": [("pipeline","/app"),("new","/app/new"),("cargo","/app/cargo")]},
    "supplier": {"email": "supplier@afrigen.africa", "tabs": [("fleet","/app"),("vault","/app/vault"),("breakdown","/app/breakdown"),("cargobids","/app/cargo")]},
    "field":    {"email": "field@afrigen.africa",    "tabs": [("audits","/app"),("border","/app/border")]},
    "admin":    {"email": "admin@afrigen.africa",    "tabs": [("overview","/app"),("verify","/app/verify"),("ledger","/app/ledger")]},
}

with sync_playwright() as p:
    browser = p.chromium.launch()

    # fresh context for auth screenshot (no session)
    ctx0 = browser.new_context(viewport={"width":1920,"height":1080})
    page0 = ctx0.new_page()
    page0.goto(BASE, wait_until="networkidle")
    page0.wait_for_timeout(1800)
    page0.screenshot(path=f"{OUT}/00-auth.png")
    print("shot 00-auth")
    ctx0.close()

    for role, cfg in roles.items():
        # brand new isolated context => no cookies, fresh auth form
        ctx = browser.new_context(viewport={"width":1920,"height":1080})
        page = ctx.new_page()
        page.goto(BASE, wait_until="networkidle")
        page.wait_for_selector('input[type="email"]', timeout=15000)
        page.locator('input[type="email"]').fill(cfg["email"])
        page.locator('input[type="password"]').fill(PASS)
        page.get_by_role("button", name="Sign in").last.click()
        page.wait_for_timeout(3000)

        for name, href in cfg["tabs"]:
            page.goto(BASE + href, wait_until="networkidle")
            page.wait_for_timeout(1800)
            page.screenshot(path=f"{OUT}/{role}-{name}.png")
            print("shot", f"{role}-{name}")

        if role == "client":
            page.goto(BASE + "/app", wait_until="networkidle")
            page.wait_for_timeout(1500)
            try:
                link = page.get_by_text("Open").or_(page.get_by_text("View")).or_(page.get_by_text("Details")).first
                link.click(timeout=3000)
                page.wait_for_timeout(2000)
                page.screenshot(path=f"{OUT}/client-contract.png")
                print("shot client-contract")
            except Exception as e:
                print("contract link skip:", e)

        ctx.close()

    browser.close()
    print("DONE")
